app/src/main/java/com/keylesspalace/tusky/StatusListActivity.kt

```diff
@@ -67,6 +80,72 @@ class StatusListActivity : BottomSheetActivity(), HasAndroidInjector {
         }
     }
 
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        val tag = hashtag
+        if (kind == Kind.TAG && tag != null) {
+            lifecycleScope.launch {
+                mastodonApi.tag(tag).fold(
+                    { tagEntity ->
+                        menuInflater.inflate(R.menu.view_hashtag_toolbar, menu)
+                        followTagItem = menu.findItem(R.id.action_follow_hashtag)
+                        unfollowTagItem = menu.findItem(R.id.action_unfollow_hashtag)
+                        followTagItem?.isVisible = tagEntity.following == false
+                        unfollowTagItem?.isVisible = tagEntity.following == true
+                        followTagItem?.setOnMenuItemClickListener { followTag() }
+                        unfollowTagItem?.setOnMenuItemClickListener { unfollowTag() }
+                    },
+                    {
+                        Log.w(TAG, "Failed to query tag #$tag", it)
+                    }
+                )
+            }
+        }
+
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    private fun followTag(): Boolean {
+        val tag = hashtag
+        if (tag != null) {
+            val context = findViewById<View>(R.id.action_follow_hashtag)
+            lifecycleScope.launch {
+                mastodonApi.followTag(tag).fold(
+                    {
+                        followTagItem?.isVisible = false
+                        unfollowTagItem?.isVisible = true
+                    },
+                    {
+                        Snackbar.make(context, getString(R.string.error_following_hashtag_format, tag), Snackbar.LENGTH_SHORT).show()
+                        Log.e(TAG, "Failed to follow #$tag", it)
+                    }
+                )
+            }
+        }
+
+        return true
+    }
+
+    private fun unfollowTag(): Boolean {
+        val tag = hashtag
+        if (tag != null) {
+            val context = findViewById<View>(R.id.action_unfollow_hashtag)
+            lifecycleScope.launch {
+                mastodonApi.unfollowTag(tag).fold(
+                    {
+                        followTagItem?.isVisible = true
+                        unfollowTagItem?.isVisible = false
+                    },
+                    {
+                        Snackbar.make(context, getString(R.string.error_unfollowing_hashtag_format, tag), Snackbar.LENGTH_SHORT).show()
```

Comment:
move the `binding` creation from `onCreate` to a class member like so `private val binding by viewBinding(ActivityStatuslistBinding::inflate)` and then you don't need the `findViewById`and can just do
`Snackbar.make(binding.root ...`

Also, it would be easy to add a retry button to the snackbar if you want.